.github/workflows/e2e-test-docker-image-build.yml

```diff
@@ -2,6 +2,12 @@ name: Create and publish a Docker image
 
 on:
   workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths:
+      - 'Dockerfile'
+      - 'terraform/**'
```

Comment:
nit: Is there a way to exclude files? Might want to specifically exclude changes to `*.md` and other similar files since some of them might have now or in the future and changes to those are not relevant. I'll leave it up to you to identify if there's room for that here

Reply:
Hmm, we could add this if the files are added in the future. Not a huge concern because these future presumably will not be edited often and running the workflow to update the images isn't really a concern